pkgs/servers/meguca/default.nix

```diff
@@ -0,0 +1,41 @@
+{ stdenv, fetchgit, cacert, go, nodejs, pkgconfig, ffmpeg-full, graphicsmagick, ghostscript, cmake, emscripten }:
+
+stdenv.mkDerivation rec {
+  name = "meguca-unstable-2018-05-05";
+
+  src = fetchgit {
+    url = "https://github.com/bakape/meguca.git";
+    rev = "4c23a96a044978533caa63943b9cf8b0a1b3df33";
+    sha256 = "1zxaya1yjalffj02dwqacw31p6grsrdbk0jyvpsv22yf44i09pzv";
+    fetchSubmodules = true;
+  };
+
+  dontUseCmakeConfigure = true;
+  enableParallelBuilding = true;
+  nativeBuildInputs = [ cmake pkgconfig ];
+  buildInputs = [ go nodejs ffmpeg-full graphicsmagick ghostscript emscripten ];
+
+  buildPhase = ''
+    mkdir -p $PWD/.home/go
+    export HOME=$PWD/.home
+    export GOPATH=$HOME/go
+    export SSL_CERT_FILE=${cacert}/etc/ssl/certs/ca-bundle.crt
+    export GIT_SSL_CAINFO=$SSL_CERT_FILE
```

Comment:
Sorry but depending on network access is a problem for inclusion in nixpkgs, since our continuous integration system will not build those packages.

Reply:
It is mainly the nodejs part of this project, that requires network access right?

Reply:
It's interesting that the darwin version built.
Is there anything I can do about this?
Do I have to redo the entire `make` process manually or is there an easier way to make this more nix-friendly?
As far as I could tell, there were never any problems with the nodejs stuff, I think it was always the Go portion that it was getting stuck on, trying to `git clone`.

Reply:
Darwin build because we currently have no sandbox support on macOS again.

Reply:
You should be able to build the server part with go2nix. It does not look like the project adds anything special to it. Be aware that you also need to run `go generate`: https://github.com/bakape/meguca/blob/master/Makefile#L63

However you will also need to build the client part, which will also run `npm install`: https://github.com/bakape/meguca/blob/master/Makefile#L31

Reply:
Ok, I'll work on the Go part for now.
I'm a bit confused as what you mean by putting the several thousand lines of nix in node-packages.
Do you mean in the node-packages directory itself, or in the meguca directory?

Reply:
Oh I see now, the readme explains what you meant.